### ISSUE TITLE
Implementation of a few deep learning layers

### ISSUE THREAD

**Issue by Rajiv2605** [mlpack/mlpack#1672]
@rcurtin @zoq I am extremely interested to work in deep learning modules. I want to implement the following layers for ANN:
- [x] Dice Loss function
- [ ] Dense blocks
- [ ] Uncertainty Map

Please let me know your thoughts. I have worked on the above layers in Caffe and Tensorflow and I want to implement them in MLPack. I have used them in biomedical dataset and they were instrumental in producing good prediction results. Please let me know what you think :smiley:

**Comment by zoq** [comment]
Hello @Rajiv2605, I think each idea has something interesting and I think each one is an opportunity to get familiar with the codebase, so if you like to work on the ideas please feel free. Perhaps it makes sense to start with the loss function.

**Comment by zoq** [comment]
Sounds good, let me know if I should clarify anything.

**Comment by Rajiv2605** [comment]
@zoq I have opened PR for dice loss class implementation here: #1674 
Please let me know what you think :)

**Comment by johnsoncarl** [comment]
@zoq I am just beginning with MLPACK. I want to take one of the above-mentioned layers. 

I am a sophomore at NIT Surat. And planning to prepare for GSOC 2019.

I have worked on ANNs in tensorflow. And is eager to learn a lot. :)

**Comment by Rajiv2605** [comment]
@johnsoncarl I am already working on those layers... So probably you can find few other layers to work on :)

**Comment by johnsoncarl** [comment]
Not a problem @Rajiv2605 I shall find some other to work on. Happy learning :)

**Comment by Rajiv2605** [comment]
@zoq @ShikharJ I've opened a PR for the dense block class: #1716 
I'll push the complete code in the subsequent commits :)

**Comment by kartikdutt18** [comment]
If it's okay I would like to take this up, since the PR for dense blocks is closed.
Thanks.
